datafusion/functions/src/regex/mod.rs

```diff
@@ -28,12 +28,53 @@ make_udf_function!(
     REGEXP_REPLACE,
     regexp_replace
 );
-export_functions!((
-    regexp_match,
-    input_arg1 input_arg2,
-    "returns a list of regular expression matches in a string. "
-),(
-    regexp_like,
-    input_arg1 input_arg2,
-    "Returns true if a has at least one match in a string,false otherwise."
-),(regexp_replace, arg1 arg2 arg3 arg4, "Replaces substrings in a string that match"));
+
+pub mod expr_fn {
+    #[doc = "returns a list of regular expression matches in a string. "]
+    #[doc = r" Return $name(arg)"]
+    pub fn regexp_match(
+        values: datafusion_expr::Expr,
+        regex: datafusion_expr::Expr,
+        flags: Option<datafusion_expr::Expr>,
+    ) -> datafusion_expr::Expr {
+        let mut args = vec![values, regex];
+        if let Some(flags) = flags {
+            args.push(flags);
+        };
+        super::regexp_match().call(args)
+    }
+
+    #[doc = "Returns true if a has at least one match in a string,false otherwise."]
+    #[doc = r" Return $name(arg)"]
+    pub fn regexp_like(
+        values: datafusion_expr::Expr,
+        regex: datafusion_expr::Expr,
+        flags: Option<datafusion_expr::Expr>,
+    ) -> datafusion_expr::Expr {
+        let mut args = vec![values, regex];
+        if let Some(flags) = flags {
+            args.push(flags);
+        };
+        super::regexp_like().call(args)
+    }
+
+    #[doc = "Replaces substrings in a string that match"]
+    #[doc = r" Return $name(arg)"]
+    pub fn regexp_replace(
+        string: datafusion_expr::Expr,
+        pattern: datafusion_expr::Expr,
+        replacement: datafusion_expr::Expr,
+        flags: Option<datafusion_expr::Expr>,
```

Comment:
Perhaps we can simplify the types of these parameters by using a use declaration.
```rust
pub mod expr_fn {
    use datafusion_expr::Expr;

    #[doc = "returns a list of regular expression matches in a string. "]
    pub fn regexp_match(values: Expr, regex: Expr, flags: Option<Expr>) -> Expr {
        let mut args = vec![values, regex];
        if let Some(flags) = flags {
            args.push(flags);
        };
        super::regexp_match().call(args)
    }
```

Reply:
Done